assets/js/components/Table/filters.jsx

```diff
@@ -30,41 +31,62 @@ const setFilter = (filters, filterKey, filterValue, filterFunction) => {
     : [...filtersList, { key: filterKey, value: filterValue, filterFunction }];
 };
 
-const getFilter = (key, list) =>
-  list.reduce(
-    (accumulator, current) =>
-      current.key === key && accumulator.length === 0
-        ? current.value
-        : accumulator,
-    []
-  );
-
 export const TableFilters = ({ config, data, filters, onChange }) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const filterFunction = (list, column) =>
+    typeof column.filter === 'function'
+      ? column.filter(list, column.key)
+      : getDefaultFilterFunction(list, column.key);
+
   return config.columns
     .filter(({ filter }) => Boolean(filter))
     .map((column) => {
-      const filterValue = getFilter(column.key, filters);
+      const [filterValue, setFilterValue] = useState([]);
+
       const filterOptions = uniq(
         data
           .map(({ [column.key]: option }) => option)
           .flat(Infinity)
           .concat(filterValue)
       );
 
+      useEffect(() => {
+        setFilterValue(searchParams.getAll(column.key));
+      }, [searchParams]);
+
+      useEffect(() => {
+        onChange(
+          setFilter(
+            filters,
+            column.key,
+            searchParams.getAll(column.key),
+            filterFunction(filterValue, column)
+          )
+        );
+      }, [JSON.stringify(filterValue)]);
```

Comment:
Just wanted to shine a bit of light on this -> https://github.com/facebook/react/issues/14476#issuecomment-471199055